File: game/game-backend/src/main/java/game/model/Grid.java
package game.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.Arrays;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Grid {
    private int[][] gridElements;
    private boolean[][] constant;
    private Score[] scores;

    public void addScore(final Score score) {
        final int length = this.scores.length;
        if (length < 5) {
            final Score[] scores = new Score[length + 1];
            scores[length] = score;
            this.scores = scores;
            System.out.println("=>success");
        } else {
            Arrays.sort(scores);
            if (score.getScore() < this.scores[0].getScore()) {
                this.scores[0] = score;
                System.out.println("=>success");
            } else {
                System.out.println("=>fail");
            }
        }
    }
}

File: game/game-frontend/src/app/model/grid.ts
import {Score} from "./score";

export class Grid {
  gridElements : number[][];
  constant : boolean[][];
  scores : Score[];

  constructor(gridElements: number[][]) {
    this.gridElements=gridElements;
    this.constant=[];
    this.scores=[];

    for(let i:number=0; i<9;i++){
      this.constant[i]=[];
      for(let j:number=0; j<9;j++){
        this.constant[i][j]=this.gridElements[i][j]!=0;
      }
    }
  }
}
